assets/kube-apiserver/kube-apiserver-deployment.yaml

```diff
@@ -250,35 +250,56 @@ spec:
       - name: portieris
         image: {{ .PortierisImage }}
         imagePullPolicy: IfNotPresent
+        command:
+          - "/portieris"
+        args:
+          - "--kubeconfig=/etc/openshift/kubeconfig/kubeconfig"
         ports:
         - containerPort: 8000
           name: http
           protocol: TCP
-        resources: {}
+{{ if .PortierisContainerResources }}
+        resources:{{ range .PortierisContainerResources }}{{ range .ResourceRequest }}
+          requests: {{ if .CPU }}
+            cpu: {{ .CPU }}{{ end }}{{ if .Memory }}
+            memory: {{ .Memory }}{{ end }}{{ end }}{{ range .ResourceLimit }}
+          limits: {{ if .CPU }}
+            cpu: {{ .CPU }}{{ end }}{{ if .Memory }}
+            memory: {{ .Memory }}{{ end }}{{ end }}{{ end }}
+{{ end }}
         terminationMessagePath: /dev/termination-log
         terminationMessagePolicy: File
+{{- if .PortierisLivenessProbe }}
+{{- $probe := .PortierisLivenessProbe }}
         livenessProbe:
           httpGet:
-            port: 8000
-            path: "/health/liveness"
-            scheme: HTTPS
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
-        readinessProbe:
+            scheme: {{ or $probe.HttpGet.Scheme "HTTPS" }}
+            port: {{ or $probe.HttpGet.Port 8000 }}
+            path: {{ or $probe.HttpGet.Path "health/liveness" }}
+          initialDelaySeconds: {{ or $probe.InitialDelaySeconds 10 }}
+          periodSeconds: {{ or $probe.PeriodSeconds 10 }}
+          timeoutSeconds: {{ or $probe.TimeoutSeconds 1 }}
+          failureThreshold: {{ or $probe.FailureThreshold 3 }}
+          successThreshold: {{ or $probe.SuccessThreshold 1 }}
+{{- else }}
+        livenessProbe:
           httpGet:
             port: 8000
-            path: "/health/readiness"
+            path: "/health/liveness"
             scheme: HTTPS
-          initialDelaySeconds: 10
-          timeoutSeconds: 10
-        env:
-        - name: KUBECONFIG
-          value: /etc/kubernetes/secret/kubeconfig
+          initialDelaySeconds: 120
+          periodSeconds: 300
+          successThreshold: 1
+          failureThreshold: 3
+          timeoutSeconds: 160
+{{ end }}
         volumeMounts:
-        - mountPath: /etc/kubernetes/secret/
-          name: secret
+        - mountPath: /etc/openshift/kubeconfig
+          name: kubeconfig
```

Comment:
You need to update the volumes to ensure this mount is available when `PortierisEnabled`.  It is only available for `ROKSMetricsImage` at the moment.